render.js

```diff
@@ -0,0 +1,28 @@
+import { Router } from "react-router-dom";
+
+export const render = (routerToComponent, rootId) => {
+  // import these here to make sure that we get the same instance used in the
+  // build of the app, rather than in the static render flow!
+  // otherwise we have 2 instances of React, and contexts etc. don't match up
+  const React = require("react");
+  const { StaticRouter } = require("react-router-dom");
+
+  if (process.env.STATIC_RENDER) {
+    const ReactDOM = require("react-dom/server");
+
+    return {
+      routerToComponent,
+      React,
+      ReactDOM,
+      StaticRouter,
+      rootId
+    };
+  } else {
+    const ReactDOM = require("react-dom");
+
+    const component = routerToComponent(Router);
+    return Promise.resolve(component).then(component =>
+      ReactDOM.hydrate(component, document.getElementById(rootId))
```

Comment:
I'm wondering if this API makes sense vs just letting the user pass in `document.getElementById(...)` like we would with `ReactDOM.render` - might be easier to keep our API closer to the React one?

Reply:
The reason it changes to be just a ID string is because it's used both to find the element when rendering, but also for _making_ the element, when static-rendering.

E.g. `document.getElementById('foo')` doesn't let the plugin know to generate a div with an ID of `foo` (unless we tried stubbing `document` or something, which I don't want to do :joy:)